File: CM.Backend/CM.Backend.Application/Models/Statistics/CattleStatisticsDto.cs
namespace CM.Backend.Application.Models.Statistics;
public class CattleStatisticsDto
{
    public List<GroupStatisticDto> GroupStatistics { get; set; } = new List<GroupStatisticDto>();
    public List<PenStatisticDto> PenStatistics { get; set; } = new List<PenStatisticDto>();
    public List<AgeGroupStatisticDto> AgeGroupStatistics { get; set; } = new List<AgeGroupStatisticDto>();
    public List<AgeGenderGroupStatisticDto> AgeGenderGroupStatisticsFemale { get; set; } = new List<AgeGenderGroupStatisticDto>();
    public List<AgeGenderGroupStatisticDto> AgeGenderGroupStatisticsMale { get; set; } = new List<AgeGenderGroupStatisticDto>();
}

public class GroupStatisticDto
{
    public int GroupId { get; set; }
    public string GroupName { get; set; }
    public int Number {  get; set; }
}

public class PenStatisticDto
{
    public int PenId { get; set; }
    public string PenName { get; set; }
    public int Number {  get; set; }
}

public class AgeGroupStatisticDto
{
    public string AgeGroupName { get; set; }
    public int Number { get; set; }
}

public class AgeGenderGroupStatisticDto
{
    public string Gender { get; set; }
    public string AgeGroupName { get; set; }
    public int Number { get; set; }
}

File: CM.Portal/src/app/api/models/cattle-statistics-dto.ts
/* tslint:disable */
/* eslint-disable */
import { AgeGenderGroupStatisticDto } from '../models/age-gender-group-statistic-dto';
import { AgeGroupStatisticDto } from '../models/age-group-statistic-dto';
import { GroupStatisticDto } from '../models/group-statistic-dto';
import { PenStatisticDto } from '../models/pen-statistic-dto';
import { RaceGroupStatisticDto } from '../models/race-group-statistic-dto';
export interface CattleStatisticsDto {
  ageGenderGroupStatisticsFemale?: Array<AgeGenderGroupStatisticDto> | null;
  ageGenderGroupStatisticsMale?: Array<AgeGenderGroupStatisticDto> | null;
  ageGroupStatistics?: Array<AgeGroupStatisticDto> | null;
  groupStatistics?: Array<GroupStatisticDto> | null;
  penStatistics?: Array<PenStatisticDto> | null;
  raceGroupStatistics?: Array<RaceGroupStatisticDto> | null;
}
